system/jhtree/jhblockcompressed.cpp

```diff
@@ -168,17 +172,23 @@ bool CJHBlockCompressedSearchNode::getKeyAt(unsigned int index, char *dst) const
 size32_t CJHBlockCompressedSearchNode::getSizeAt(unsigned int index) const
 {
     if (keyHdr->hasSpecialFileposition())
-        return keyLen + sizeof(offset_t);
+    {
+        if (hasFilePosition)
+            return keyLen + sizeof(offset_t);
```

Comment:
I think this change is wrong - this is the size as seen by the consumer, which should include the zeroed fileposition